src/api/app/models/package.rb

```diff
@@ -428,8 +428,9 @@ def sources_changed(opts = {})
   end
 
   def self.source_path(project, package, file = nil, opts = {})
-    path = "/source/#{URI.escape(project)}/#{URI.escape(package)}"
-    path += "/#{URI.escape(file)}" if file.present?
+    path = "/source/#{project}/#{package}"
+    path = Addressable::URI.escape(path)
+    path += "/#{CGI.escape(file)}" if file.present?
```

Comment:
In other changes from this PR, `URI.escape` was changed to `Addressable::URI.escape`. Is there a reason here for not doing the same?

Reply:
Anchors are a valid part of URIs.

```ruby
[6] pry(#<SourceController>)> Addressable::URI.escape('bnc#620675.diff')
=> "bnc#620675.diff"
[6] pry(#<SourceController>)> CGI.escape('bnc#620675.diff')
=> "bnc%23620675.diff"
```

Reply:
`file` is from `params` which is unescaped by RoR. We need to escape it again to send to the backend.

Reply:
Sure about anchors. I just wonder why it's not like this:

```suggestion
    path += "/#{Adresssable::URI.escape(file)}" if file.present?
```

Reply:
Alright, thanks for the output in your previous comment. I didn't check before how those behave differently.

Reply:
Yes, we found out that in this case `Addressable::URI.escape` was not escaping some characters that we need to be escaped... For example a `#` in the middle of a package name.